activesupport/lib/active_support/error_reporter.rb

```diff
@@ -108,13 +112,39 @@ def handle(*error_classes, severity: :warning, context: {}, fallback: nil, sourc
     #   source of the error. Subscribers can use this value to ignore certain
     #   errors. Defaults to <tt>"application"</tt>.
     def record(*error_classes, severity: :error, context: {}, source: DEFAULT_SOURCE)
-      error_classes = [StandardError] if error_classes.blank?
+      error_classes = DEFAULT_RESCUE if error_classes.empty?
       yield
     rescue *error_classes => error
       report(error, handled: false, severity: severity, context: context, source: source)
       raise
     end
 
+    # Either report the given error when in production, or raise it when in development or test.
+    #
+    # When raising, the original error is wrapped in a different error class to ensure
+    # it's not being rescued higher in the stack and will be surfaced to the developer.
+    #
+    # This method is intended for reporting violated assertions about preconditions, or similar
+    # cases that can and should be gracefully handled in production, but that aren't supposed to happen.
+    #
+    #   example:
+    #
+    #     def edit
+    #       if published?
+    #         Rails.error.report(RuntimeError.new("[BUG] Attempting to edit a published article, that shouldn't be possible."))
```

Comment:
I guess this should be calling `unexpected`.

Feels not-ideal for the example to involve (ugly, and backtrace-missing) manual exception construction... but I don't really have a compelling alternative to suggest.

Reply:
I tweaked the backtrace handling. Let me know what you think now.